trino/sqlalchemy/datatype.py

```diff
@@ -193,7 +211,9 @@ def parse_sqltype(type_str: str) -> TypeEngine:
     type_class = _type_map[type_name]
     type_args = [int(o.strip()) for o in type_opts.split(",")] if type_opts else []
     if type_name in ("time", "timestamp"):
-        type_kwargs = dict(timezone=type_str.endswith("with time zone"))
-        # TODO: support parametric timestamps (https://github.com/trinodb/trino-python-client/issues/107)
-        return type_class(**type_kwargs)
+        if type_str.endswith("with time zone"):
+            type_kwargs: Dict[str, Any] = dict(timezone=True)
+            if type_opts is not None:
+                type_kwargs["precision"] = int(type_opts)
+            return type_class(**type_kwargs)
```

Comment:
I expect this `if` block handle all cases of `time` & `timestamp` datatype:
```python
    if type_name in ("time", "timestamp"):
        type_kwargs: Dict[str, Any] = dict()
        if type_str.endswith("with time zone"):
            type_kwargs["timezone"] = True
        if type_opts is not None:
            type_kwargs["precision"] = int(type_opts)
        return type_class(**type_kwargs)
```

Reply:
done